stdlib/builtins.pyi

```diff
@@ -196,9 +198,9 @@ _LiteralInteger = _PositiveInteger | _NegativeInteger | Literal[0]  # noqa: Y026
 
 class int:
     @overload
-    def __new__(cls: type[Self], __x: str | ReadableBuffer | SupportsInt | SupportsIndex | SupportsTrunc = ...) -> Self: ...
+    def __new__(cls, __x: str | ReadableBuffer | SupportsInt | SupportsIndex | SupportsTrunc = ...) -> Self: ...
```

Comment:
I noticed this change has introduced a new pyright issue at  https://github.com/python/typeshed/blob/main/scripts/stubsabot.py#L37

Reply:
My guess is that's probably a pyright issue in how it handles `Self`. What's the error message pyright produces?

Reply:
It thinks `member` is a `int` when previously it saw it as a `ActionLevelSelf`
![image](https://user-images.githubusercontent.com/1350584/220212745-30db2270-de2f-4b56-be0e-1cfa78cf812a.png)
![image](https://user-images.githubusercontent.com/1350584/220212787-5f916dd8-c622-444d-8526-9cc9357e4671.png)

vs before

![image](https://user-images.githubusercontent.com/1350584/220213050-a19ffc46-3cc5-4ea2-9cae-0b92a5aec21f.png)

I guess pyright doesn't take into account that the first classmethod parameter (`cls`) could be something else than it's own class.
mypy does it correctly:
![image](https://user-images.githubusercontent.com/1350584/220213125-52111878-6219-4857-887a-57588f6bbad6.png)

Reply:
Hmm, yeah that seems like it might be a pyright bug (though Eric will be the judge of that). If you file an issue over there, he'll probably either fix it or triage it as "wontfix" within 24 hours :D

We also don't need the `ActionLevelSelf` `TypeVar` in stubsabot anymore, though, we can just use `typing_extensions.Self` there.

Reply:
> We also don't need the `ActionLevelSelf` `TypeVar` in stubsabot anymore, though, we can just use `typing_extensions.Self` there.

- #9785

Reply:
Alright, created an issue here: https://github.com/microsoft/pyright/issues/4668